miner/taiko_worker.go

```diff
@@ -72,39 +77,37 @@ func (w *Miner) buildTransactionsLists(
 		localTxs, remoteTxs = w.getPendingTxs(localAccounts, baseFee)
 	)
 
-	commitTxs := func(firstTransaction *types.Transaction) (*types.Transaction, *PreBuiltTxList, error) {
+	commitTxs := func(presetTxs []*types.Transaction) ([]*types.Transaction, *PreBuiltTxList, error) {
 		env.tcount = 0
 		env.txs = []*types.Transaction{}
 		env.gasPool = new(core.GasPool).AddGas(blockMaxGasLimit)
 		env.header.GasLimit = blockMaxGasLimit
 
-		lastTransaction := w.commitL2Transactions(
+		txsPruningResult, err := w.commitL2Transactions(
 			env,
-			firstTransaction,
+			presetTxs,
 			newTransactionsByPriceAndNonce(signer, maps.Clone(localTxs), baseFee),
 			newTransactionsByPriceAndNonce(signer, maps.Clone(remoteTxs), baseFee),
 			maxBytesPerTxList,
 			minTip,
 		)
-
-		b, err := encodeAndCompressTxList(env.txs)
 		if err != nil {
 			return nil, nil, err
 		}
 
-		return lastTransaction, &PreBuiltTxList{
-			TxList:           env.txs,
+		return txsPruningResult.Pruned, &PreBuiltTxList{
+			TxList:           txsPruningResult.Remaining,
 			EstimatedGasUsed: env.header.GasLimit - env.gasPool.Gas(),
```

Comment:
`EstimatedGasUsed ` will likely bigger than real value, due to double complete `N%100` txs.

Reply:
Yes its ok atm.